dali/operators/decoder/video/video_decoder_base.h

```diff
@@ -60,19 +62,64 @@ class DLL_PUBLIC VideoDecoderBase {
     return shape;
   }
 
+
   /**
-   * @brief Decode sample with index `idx` to `output` tensor.
+   * Decode given number of frames from a FramesDecoder.
+   *
+   * This function allows optional padding of the output data, when
+   * there's not enough frames inside FramesDecoder to fill the `num_frames` specified.
+   *
+   * The `pad_value` argument determines, with what the VideoDecoder shall pad
+   * the partial sequence. The tensor passed to the `pad_value` shall be an entire
+   * frame. This frame will be repeated at the end of partial sequence.
+   * If the `pad_value` is not provided, the padding will not happen and the DecodeFrames
+   * will return a partial sequence.
+   *
+   * @param output The SampleView in which the decoded sequence will be put.
+   * @param sample_idx Index of the encoded video that shall be decoded.
+   * @param num_frames How many frames shall be decoded.
+   * @param pad_value What to (optionally) pad the partial sequence with.
+   * @return False, if less than `num_frames` have been decoded.
    */
-  void DecodeSample(SampleView<OutBackend> output, int64_t idx) {
-    auto &frames_decoder = *frames_decoders_[idx];
-    int64_t num_frames = output.shape()[0];
+  bool DecodeFrames(SampleView<OutBackend> output, int64_t sample_idx, int64_t num_frames,
+                    std::optional<SampleView<OutBackend>> pad_value = std::nullopt,
+                    cudaStream_t stream = 0) {
```

Comment:
I'd recommend swapping the two arguments and leaving the `stream` without a default value - this will help us avoid accidentally decoding stuff on stream 0.

Reply:
Another take: use `std::optional<cudaStream_t>` and keep the default as `nullopt` - the CPU variant will try to obtain the stream and fail fast in case where the value wasn't supplied.

Reply:
Changed to `std::optional<cudaStream_t>`